opslevel/resource_opslevel_check_repository_grep.go

```diff
@@ -1,108 +1,242 @@
+		return
+	}
+
+	stateModel, diags := NewCheckRepositoryGrepResourceModel(ctx, *data)
+	stateModel.EnableOn = planModel.EnableOn
+	// Custom Prop Overrides Here
+	stateModel.LastUpdated = timeLastUpdated()
+	resp.Diagnostics.Append(diags...)
+
+	tflog.Trace(ctx, "created a check repositorygrep resource")
+	resp.Diagnostics.Append(resp.State.Set(ctx, &stateModel)...)
+}
+
+func (r *CheckRepositoryGrepResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var planModel CheckRepositoryGrepResourceModel
+
+	// Read Terraform prior state data into the planModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &planModel)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	data, err := r.client.GetCheck(asID(planModel.Id))
+	if err != nil {
+		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("Unable to read check repositorygrep, got error: %s", err))
+		return
+	}
+	stateModel, diags := NewCheckRepositoryGrepResourceModel(ctx, *data)
+	resp.Diagnostics.Append(diags...)
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &stateModel)...)
+}
+
+func (r *CheckRepositoryGrepResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var planModel CheckRepositoryGrepResourceModel
+
+	// Read Terraform plan data into the planModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &planModel)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	enabledOn, err := iso8601.ParseString(planModel.EnableOn.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("error", err.Error())
+		return
+	}
+	input := opslevel.CheckRepositoryGrepUpdateInput{
+		CategoryId: opslevel.RefOf(asID(planModel.Category)),
+		Enabled:    planModel.Enabled.ValueBoolPointer(),
+		EnableOn:   &iso8601.Time{Time: enabledOn},
+		FilterId:   opslevel.RefOf(asID(planModel.Filter)),
+		LevelId:    opslevel.RefOf(asID(planModel.Level)),
+		Id:         asID(planModel.Id),
+		Name:       opslevel.RefOf(planModel.Name.ValueString()),
+		Notes:      planModel.Notes.ValueStringPointer(),
+		OwnerId:    opslevel.RefOf(asID(planModel.Owner)),
+	}
+
+	input.DirectorySearch = planModel.DirectorySearch.ValueBoolPointer()
+	resp.Diagnostics.Append(planModel.Filepaths.ElementsAs(ctx, &input.FilePaths, false)...)
+	if planModel.FileContentsPredicate != nil {
+		input.FileContentsPredicate = &opslevel.PredicateUpdateInput{
+			Type:  opslevel.RefOf(opslevel.PredicateTypeEnum(planModel.FileContentsPredicate.Type.String())),
+			Value: opslevel.RefOf(planModel.FileContentsPredicate.Value.String()),
+		}
+	}
+
+	data, err := r.client.UpdateCheckRepositoryGrep(input)
+	if err != nil {
+		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("Unable to update check_repository_grep, got error: %s", err))
+		return
+	}
+
+	stateModel, diags := NewCheckRepositoryGrepResourceModel(ctx, *data)
+	stateModel.EnableOn = planModel.EnableOn
```

Comment:
Why does this have to be set here?

Reply:
Because of a time formatting problem otherwise it creates an infinite plan / apply situation.

If the customer specifies `2022-05-23T14:14:18.782000Z` the api will return `2022-05-23T14:14:18Z` and if we store directly into the state that - we get an infinite apply.

So this basically override what goes into the state to be exactly what the customer set because terraform does a diff of the plan and the state after calling the create or update functions.